server/store/sqlstore/store.go

```diff
@@ -981,6 +995,81 @@ func (s *SQLStore) IsUserPresentInWhitelist(userID string) (bool, error) {
 	return result != "", nil
 }
 
+func (s *SQLStore) StoreInvitedUser(invitedUser *storemodels.InvitedUser) error {
+	pendingSince := invitedUser.InvitePendingSince.UnixMicro()
+	lastSentAt := invitedUser.InviteLastSentAt.UnixMicro()
+
+	query := s.getQueryBuilder().
+		Insert(invitedUsersTableName).
+		Columns("mmUserID", "invitePendingSince", "inviteLastSentAt").
+		Values(invitedUser.ID, pendingSince, lastSentAt).
+		SuffixExpr(sq.Expr("ON CONFLICT (mmUserID) DO UPDATE SET invitePendingSince = ?, inviteLastSentAt = ?", pendingSince, lastSentAt))
+
+	if _, err := query.Exec(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *SQLStore) GetInvitedUser(mmUserID string) (*storemodels.InvitedUser, error) {
+	query := s.getQueryBuilder().
+		Select("mmUserID", "invitePendingSince", "inviteLastSentAt").
+		From(invitedUsersTableName).
+		Where(sq.Eq{"mmUserID": mmUserID})
+
+	rows, err := query.Query()
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	var result *storemodels.InvitedUser
+	if rows.Next() {
+		var id string
+		var pendingSince int64
+		var lastSentAt int64
+
+		if scanErr := rows.Scan(&id, &pendingSince, &lastSentAt); scanErr != nil {
+			return nil, scanErr
+		}
+
+		result = &storemodels.InvitedUser{
+			ID:                 id,
+			InvitePendingSince: time.UnixMicro(pendingSince),
+			InviteLastSentAt:   time.UnixMicro(pendingSince),
+		}
+	}
+
+	return result, nil
+}
+
+func (s *SQLStore) DeleteUserInvite(mmUserID string) error {
+	if _, err := s.getQueryBuilder().Delete(invitedUsersTableName).Where(sq.Eq{"mmUserID": mmUserID}).Exec(); err != nil {
+		return err
+	}
+
+	return nil
+}
```

Comment:
I wonder if should instead mark this as `DeleteAt = NOW()`, with the idea being that we preserve this metadata and possibly also use it to decide if we resend to a user that was previously invited?

Reply:
That is indeed better, but I'm partially opposed to it as it would introduce `include_deleted` complexity, and it may cause mis-matched semantics vs the whitelist store methods.

I also don't see us needing to keep the exact timestamp the invite was originally sent because the invite is preserved until first connect, after which we have the whitelist.

Reading between the lines, I think `OnDisconnectedTokenHandler` handles that sort of flow better than resending an initial connection invite.

Reply:
Sounds good, especially given the whitelist comment you shared earlier preventing re-invites.